src/vorta/views/source_tab.py

```diff
@@ -20,17 +39,52 @@ def __init__(self, parent=None):
         self.excludePatternsField.textChanged.connect(self.save_exclude_patterns)
         self.excludeIfPresentField.textChanged.connect(self.save_exclude_if_present)
         self.populate_from_profile()
+        
+    def set_path_info(self,path,data_size,files_count):
+        items = self.sourceFilesWidget.findItems(path,QtCore.Qt.MatchExactly)
+        for item in items:
+            self.sourceFilesWidget.item(item.row(),2).setText(pretty_bytes(data_size))
+            self.sourceFilesWidget.item(item.row(),3).setText(format(files_count))
+            db_item = SourceFileModel.get(dir=path)
+            db_item.dir_size = data_size
+            db_item.dir_files_count = files_count
+            db_item.save()
+            
+
+    def add_source_to_table(self,source,update_data):
+        indexRow = self.sourceFilesWidget.rowCount()
+        self.sourceFilesWidget.insertRow(indexRow)
+        itemPath = QTableWidgetItem(source.dir)
+        self.sourceFilesWidget.setItem(indexRow,0,itemPath)
+        if source.path_isdir == True:
+            self.sourceFilesWidget.setItem(indexRow,1,QTableWidgetItem("<DIR>"))
+        else:
+            self.sourceFilesWidget.setItem(indexRow,1,QTableWidgetItem("<File>"))
+        if update_data == True:
+            self.sourceFilesWidget.setItem(indexRow,2,QTableWidgetItem("load..."))
+            self.sourceFilesWidget.setItem(indexRow,3,QTableWidgetItem("load..."))
+            getDir = FilePathInfoAsync(source.dir)
+            getDir.signal.connect(self.set_path_info)
+            self.updateThreads.append(getDir) # this is ugly, is there a better way to keep the thread object?
```

Comment:
Is there a better way to prevent the thread object from destruction, than adding it to the array?